src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.Editor.cs

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.MoveType
+{
+    internal abstract partial class AbstractMoveTypeService<TService, TTypeDeclarationSyntax, TNamespaceDeclarationSyntax, TMemberDeclarationSyntax, TCompilationUnitSyntax>
+    {
+        /// <summary>
+        /// We perform 3 different kinds of edits for Move Type Code Action
+        /// 1. Rename file to match type
+        /// 2. Rename type to match file
+        /// 3. Move type to a new file
+        /// </summary>
+        private partial class Editor
```

Comment:
(cyrus): So, IMO, this should be an abstact class with at least one abstrcat method like "GetOperationsAsync" on it.  Then we'd have multiple subclasses for the different edits.  
THe base class would share only the state truly common to them all (like cancellatiohn token).  Anything specific to one would be in that subclass.